src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -5,37 +5,40 @@
 import core.mate.academy.model.Excavator;
 import core.mate.academy.model.ExcavatorProducer;
 import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
 import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+public class MachineServiceImpl implements MachineService<Machine> {
     @Override
     public List<Machine> getAll(Class<? extends Machine> type) {
-        MachineProducer<? extends Machine> mp;
         if (type == Bulldozer.class) {
-            mp = new BulldozerProducer();
+            return new ArrayList<>(new BulldozerProducer().get());
         } else if (type == Excavator.class) {
```

Comment:
do we need else block after return statement?

Reply:
Fair enough!)